stdlib/contextlib.pyi

```diff
@@ -74,6 +74,12 @@ class _GeneratorContextManager(AbstractContextManager[_T_co, bool | None], Conte
             self, type: type[BaseException] | None, value: BaseException | None, traceback: TracebackType | None
         ) -> bool | None: ...
 
+@overload
+def contextmanager(func: Callable[_P, Generator[_T_co]]) -> Callable[_P, _GeneratorContextManager[_T_co]]: ...
+@overload
+@deprecated(
+    "Returning only an 'Iterator' from a function decorated with 'contextmanager' is deprecated. Use generator functions and 'Generator' instead."
```

Comment:
Maybe an error message like this?

```suggestion
    "Using `-> Iterator[Foo]` with `@contextmanager` is deprecated. Use `-> Generator[Foo]` instead."
```

IMO this error is easier to fix, especially for users who are not confident with the details of Python's features.

Reply:
That phrasing does seem more actionable. I was trying to account for circumstances where the return type isn't written explicitly but instead inferred; maybe the process of annotating the return type would lead to the source of the issue being noticed properly by the user and fixed anyway, then.